### PR TITLE
ender's 1.12.2 FullMappingAlchemist attempt

### PR DESCRIPTION
**Pull request by ENDERZOMBI102**
This PR focuses on two aspects of the game:
- [ ] UI-related classes
	- [ ] Screens
	- [ ] Widgets
	- [ ] Realms stuff
	- [ ] Toasts
- [ ] Sound-related classes
	- [ ] SoundEngine
	- [x] Sound categories
	- [ ] `static final Sound` fields

### CODE REVIEW

File: mappings/net/minecraft/entity/player/PlayerInventory.mapping [hunk 1]
```diff
@@ -7,6 +11,25 @@ CLASS net/minecraft/class_987 net/minecraft/entity/player/PlayerInventory
...
+	METHOD method_11232 (Lnet/minecraft/class_1069;IILnet/minecraft/class_322;)I
+		ARG 1 item
+		ARG 4 nbt
+	METHOD method_13250 (Lnet/minecraft/class_1071;)V
+		ARG 1 stack
+	METHOD method_13252 (Lnet/minecraft/class_2232;)F
+		ARG 1 state
+	METHOD method_13253 (Lnet/minecraft/class_1071;)I
+		ARG 1 stack
+	METHOD method_13254 stacksAreEqual (Lnet/minecraft/class_1071;Lnet/minecraft/class_1071;)Z
```
Comment (BleachDev) [inline mappings/net/minecraft/entity/player/PlayerInventory.mapping:23]: `areStacksEqual` or `areEqual`, `stacksAreEqual` sounds pretty awkward

File: mappings/net/minecraft/item/Item.mapping [hunk 2]
```diff
@@ -406,7 +407,7 @@ CLASS net/minecraft/class_1069 net/minecraft/item/Item
 		COMMENT @see net.minecraft.item.ArmorItem
 		COMMENT @return Enchantability
 	METHOD method_3363 getMeta (I)I
-		ARG 1 i
+		ARG 1 meta
```
Comment (BleachDev) [inline mappings/net/minecraft/item/Item.mapping:410]: getMeta(meta) is a bit sus

File: mappings/net/minecraft/item/VariantBlockItem.mapping [hunk 3]
```diff
@@ -1,7 +1,19 @@
...
+	FIELD field_15114 translationProvider Lnet/minecraft/class_1102$class_3057;
 	FIELD field_9148 variantBlock Lnet/minecraft/class_197;
 	FIELD field_9149 nameFunction Lcom/google/common/base/Function;
 	METHOD <init> (Lnet/minecraft/class_197;Lnet/minecraft/class_197;Lcom/google/common/base/Function;)V
 		ARG 1 block
 		ARG 2 variantBlock
 		ARG 3 nameFunction
+	METHOD <init> (Lnet/minecraft/class_197;Lnet/minecraft/class_197;Lnet/minecraft/class_1102$class_3057;)V
+		ARG 1 block
+		ARG 2 variantblock
```
Comment (BleachDev) [inline mappings/net/minecraft/item/VariantBlockItem.mapping:11]:
```suggestion
		ARG 2 variantBlock
```

File: mappings/net/minecraft/realms/RealmsClickableScrolledSelectionList.mapping [hunk 4]
```diff
@@ -6,6 +6,12 @@ CLASS net/minecraft/realms/RealmsClickableScrolledSelectionList net/minecraft/re
 		ARG 3 top
 		ARG 4 bottom
 		ARG 5 entryHeight
+	METHOD customMouseEvent (IIIFI)V
+		ARG 1 yStart
+		ARG 2 yEnd
```
Comment (BleachDev) [inline mappings/net/minecraft/realms/RealmsClickableScrolledSelectionList.mapping:11]:
```suggestion
		ARG 1 startY
		ARG 2 endY
```
Naming conventions

File: mappings/net/minecraft/sound/BlockSoundGroup.mapping [hunk 5]
```diff
@@ -1,15 +1,43 @@
...
+	FIELD field_506 stepSound Lnet/minecraft/class_2926;
 	FIELD field_507 volume F
 	FIELD field_508 pitch F
+	METHOD <init> (FFLnet/minecraft/class_2926;Lnet/minecraft/class_2926;Lnet/minecraft/class_2926;Lnet/minecraft/class_2926;Lnet/minecraft/class_2926;)V
+		ARG 1 volume
+		ARG 2 pitch
+		ARG 3 breakSOund
+		ARG 4 stepSound
+		ARG 5 placeSound
+		ARG 6 hitSOund
```
Comment (BleachDev) [inline mappings/net/minecraft/sound/BlockSoundGroup.mapping:28]:
```suggestion
		ARG 3 breakSound
		ARG 4 stepSound
		ARG 5 placeSound
		ARG 6 hitSound
```

File: mappings/net/minecraft/sound/BlockSoundGroup.mapping [hunk 6]
```diff
@@ -1,15 +1,43 @@
...
 	FIELD field_12762 STONE Lnet/minecraft/class_200;
+	FIELD field_12763 METAL Lnet/minecraft/class_200;
+	FIELD field_12764 GLASS Lnet/minecraft/class_200;
+	FIELD field_12765 CLOTH Lnet/minecraft/class_200;
+	FIELD field_12766 SAND Lnet/minecraft/class_200;
+	FIELD field_12767 SNOW Lnet/minecraft/class_200;
+	FIELD field_12768 LADDER Lnet/minecraft/class_200;
+	FIELD field_12769 ANVIL Lnet/minecraft/class_200;
+	FIELD field_12770 SLIME Lnet/minecraft/class_200;
+	FIELD field_12771 breakSOund Lnet/minecraft/class_2926;
```
Comment (BleachDev) [inline mappings/net/minecraft/sound/BlockSoundGroup.mapping:14]:
```suggestion
	FIELD field_12771 breakSound Lnet/minecraft/class_2926;
```

File: mappings/net/minecraft/BlockItem.mapping [hunk 7]
```diff
@@ -0,0 +1,4 @@
+CLASS net/minecraft/class_3056 net/minecraft/BlockItem
```
Comment (arthurbambou) [inline mappings/net/minecraft/BlockItem.mapping:1]: Move in item package

File: mappings/net/minecraft/MagmaBlock.mapping [hunk 8]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/class_3010 net/minecraft/MagmaBlock
```
Comment (arthurbambou) [inline mappings/net/minecraft/MagmaBlock.mapping:1]: Move in block package

File: mappings/net/minecraft/entity/player/PlayerInventory.mapping (hunk 1 above)
Comment (ENDERZOMBI102) [reply mappings/net/minecraft/entity/player/PlayerInventory.mapping:23]: `areStacksEqual`

File: mappings/net/minecraft/item/Item.mapping (hunk 2 above)
Comment (ENDERZOMBI102) [reply mappings/net/minecraft/item/Item.mapping:410]: reset to obfuscated

File: mappings/net/minecraft/realms/RealmsClickableScrolledSelectionList.mapping (hunk 4 above)
Comment (ENDERZOMBI102) [reply mappings/net/minecraft/realms/RealmsClickableScrolledSelectionList.mapping:11]: fixed the typo

File: mappings/net/minecraft/item/VariantBlockItem.mapping (hunk 3 above)
Comment (ENDERZOMBI102) [reply mappings/net/minecraft/item/VariantBlockItem.mapping:11]: fixed the typo

File: mappings/net/minecraft/realms/RealmsClickableScrolledSelectionList.mapping (hunk 4 above)
Comment (ENDERZOMBI102) [reply mappings/net/minecraft/realms/RealmsClickableScrolledSelectionList.mapping:11]: done

File: mappings/net/minecraft/sound/BlockSoundGroup.mapping (hunk 5 above)
Comment (ENDERZOMBI102) [reply mappings/net/minecraft/sound/BlockSoundGroup.mapping:28]: typo fixed

**Review by BoogieMonster1O1 — CHANGES_REQUESTED** [review]
pls no large yarn prs. split into smaller ones
- easier to review
- no merge conflicts

**Comment by BoogieMonster1O1** [comment]
what must happen to this pr

**Comment by arthurbambou** [comment]
Idk, it's still a draft

**Pull request closed without merging**